### PR TITLE
Fixes two incorrect english strings

### PR DESCRIPTION
**Pull request by jonasheinrich**
https://community.openproject.org/work_packages/20910

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://jonasheinrich-fix-two-incorrect-english-strings.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://jonasheinrich-fix-two-incorrect-english-strings.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/models/project_association.rb [hunk 1]
```diff
@@ -37,7 +37,8 @@ class ProjectAssociation < ActiveRecord::Base
   belongs_to :project_b, class_name:  'Project',
                          foreign_key: 'project_b_id'
 
-  validates_presence_of :project_a, :project_b
+  alias_attribute :dependent_project, :project_b
+  validates_presence_of :project_a, :dependent_project
```
Comment (NobodysNightmare) [inline app/models/project_association.rb]: 1. probably @houndci knows things that we should take into account
2. If I understand this correctly you are aliasing `project_b` into `dependent` project to be able to give it a nice translation. Why don't you  do the same for `project_a`?

**Comment by NobodysNightmare** [comment]
A test was broken due to the renaming:

```
Failure/Error: expect(project_association.errors[:project_b]).to eq(["can't be blank"])

       expected: ["can't be blank"]
            got: []
```

File: config/locales/en.yml [hunk 2]
```diff
@@ -221,7 +221,7 @@ en:
         could_not_be_copied: "could not be (fully) copied"
       models:
         project_association:
-          identical_projects: "can not be created from one project to itself"
+          identical_projects: "Dependency cannot be created between one project and itself."
```
Comment (ulferts) [inline config/locales/en.yml:225]: :+1: for diverging from the string specified in the WP.

File: app/models/project_association.rb (hunk 1 above)
Comment (ulferts) [inline app/models/project_association.rb]: I think you should either rename the `belongs_to :project_b` directly, if it is always the dependent project (and do the same for `:project_a`) or only rely on I18n to output the human attribute name for the keys `:project_a` and `:project_b`

**Comment by ulferts** [comment]
@jonasheinrich can you please take another look at this one so we can process it further?

**Pull request merged**